### ISSUE TITLE
Multiple users on macOS

### ISSUE THREAD

**Issue by nudefireninja** [federico-terzi/espanso#721]
### Bug

The daemon doesn't start correctly when attempting to start espanso in a second user account of the system.

<img width="487" alt="espanso-daemon-bug" src="https://user-images.githubusercontent.com/6565746/127692886-7cf5511a-b4c9-4031-a310-946f37ee634f.png">

Steps to reproduce:

1. Boot macOS, log in to first user account.
2. Open the fast user switching menu and log into a second user account.
3. Try to start espanso.

**Expected result:** two independent espanso processes for each user.

**Actual result:** espanso doesn't start on subsequent logins. Only the first user to log in can use espanso.

When checking Activity Monitor I can see two espanso processes, both belonging to the first user.

The login order doesn't matter. The first login can use espanso, others cannot.

**Comment by federico-terzi** [comment]
Hey @nudefireninja,

Thanks for reporting this issue! This seems like a bug indeed, could you please post the output of `espanso log` here?

Cheers :)

**Comment by nudefireninja** [comment]
Okay, here is the log of the first account, with espanso successfully launched after login:

[espanso-log-family.txt](https://github.com/federico-terzi/espanso/files/6913463/espanso-log-family.txt)

Then I switched back to my personal account and ran `espanso start`, and here is the log after that:

[espanso-log-tem.txt](https://github.com/federico-terzi/espanso/files/6913466/espanso-log-tem.txt)

Looking at the timestamps it seems no new log messages were written by the failed start.

Thanks for looking into this!

**Comment by federico-terzi** [comment]
@nudefireninja thank you! I'll investigate. I don't have any ETA yet, but hopefully will be able to figure out a solution :)

**Comment by greatestview** [comment]
Same issue here, also two espanso processes, both belonging to the first user. +1

**Comment by zambonic** [comment]
I found this searching for my own issue which is identical except it happens on Windows 11. Perhaps for the same reasons?

I installed Espanso "for all users" and can see/run in either of my two user accounts, however Espanso will only actually run for whichever user launches it first. Since it runs on startup, that means whichever user logs in first. If the second user tries to run Espanso, nothing happens.

**Comment by smeech** [comment]
Is this still a problem with the current version of Espanso?

**Comment by nudefireninja** [comment]
Yeah (version 2.2.1), and now I get this when I run `espanso start` on the second account:

```
unable to start service: timed out
Hint: sometimes this happens because another Espanso process is left running for some reason.
      Please try running 'espanso restart' or manually killing all Espanso processes, then try again.
```

**Comment by smeech** [comment]
Thank you for updating us. Now we have more people involved we may get around to some of these older Issues.
For information, are you able to list all the Espanso instances, and kill the ones for the other account, and does it work after that?

**Comment by nudefireninja** [comment]
Thanks for looking into it!

There are three "espanso" processes when espanso is running on one user (no longer two as when I originally reported this issue). When I run `espanso start` on the second user, there is a fourth process, but after around 5 seconds it quits and I get the aforementioned error message.

If I quit any of the three espanso processes as the second user with Activity Monitor (having to authenticate as the first user to do so), then the other two processes quit as well.

Then if I try to run `espanso start` (still as the second user) I get the same thing (error message, no espanso).

I tried a few different things and still could not get espanso to start on the second user. If I reboot the system then I can start espanso and use it on whichever account I use first.

---

During my investigation I also found a couple oddities.

<details>

<summary>Multiple espansos in $PATH</summary>

One is that if I run `type -a espanso` I get the following:

```
espanso is /opt/homebrew/bin/espanso
espanso is /usr/local/bin/espanso
```

Which is a little unexpected. I think I originally installed espanso by downloading the .zip and moving the application to the /Applications folder, but more recently I installed it via Homebrew, and I can't remember if I uninstalled the original Espanso before installing with Homebrew, but my guess is that the `/usr/local/bin/espanso` is a remnant from that.

Both of them are actually symlinks to the executable in /Applications:

```
lrwxr-xr-x  1 tem   admin    48B 18 Nov  2022 /opt/homebrew/bin/espanso -> /Applications/Espanso.app/Contents/MacOS/espanso
lrwxr-xr-x  1 root  wheel    48B 21 Mar  2022 /usr/local/bin/espanso -> /Applications/Espanso.app/Contents/MacOS/espanso
```

Anyway, that's probably not causing any issues right now.

</details>

---

<details>

<summary>Different login items entries</summary>

The second oddity is that when I checked login items on both users, they had a different entry for espanso under "Allow in the Background".

On the first account:

<img width="474" alt="login-items-tem" src="https://github.com/espanso/espanso/assets/6565746/f0eae9d0-1215-42cb-b66c-9c52a1cb3285">

On the second account:

<img width="477" alt="login-items-family" src="https://github.com/espanso/espanso/assets/6565746/a2bfc36f-59da-4a15-9e1a-8db750c23df6">

The second one lets me reveal it in the Finder at /opt/homebrew/bin/espanso. (The different switch colors are due to the contrast/transparency accessibility settings on the first account, just in case you were wondering.)

I tried disabling both and it made no difference -- I still had to reboot in order to get espanso running on the second user.

</details>

By the way, I updated to macOS Sonoma 14.3.1 (the latest non-beta version) before I did anything.

**Comment by smeech** [comment]
Is #919 any help?

**Comment by nudefireninja** [comment]
I could not find a way to try portable mode because I guess that's only for Windows. Also if I try to run `sudo espanso start` I'm not able to authenticate with my non-admin password, so I tried first switching to the admin user with `su tem` but then it says "espanso is already running".

However, while looking for the portable mode I discovered the `--unmanaged` option and when I tried `espanso start --unmanaged` it actually worked! The first time it showed the configuration backup+migration, and I was able to use espanso triggers on both accounts just by switching, not having to do anything extra.

It still won't start automatically on login, even though `espanso service check` does say that it's registered (`espanso service start` fails in the same way as `espanso start`).

**Comment by smeech** [comment]
I'm tempted to close this as we have a workaround for this edge-case, and it's unlikely to be addressed in the coding? Happy to have it reopened if necessary.
